File: 1657/1657.go
package main

import (
    "fmt"
    "sort"
)

func closeStrings(word1, word2 string) bool {
    if len(word1) != len(word2) {
        return false
    }

    count1 := make(map[rune]int)
    count2 := make(map[rune]int)

    for _, char := range word1 {
        count1[char]++
    }
    for _, char := range word2 {
        count2[char]++
    }

    if len(count1) != len(count2) {
        return false
    }

    var count1Values, count2Values []int
    for _, value := range count1 {
        count1Values = append(count1Values, value)
    }
    for _, value := range count2 {
        count2Values = append(count2Values, value)

File: Solutions/1657/1657.js
function closeStrings(word1, word2) {
  // Check for length equality
  if (word1.length !== word2.length) {
    return false;
  }

  // Create character counters
  const count1 = new Map();
  const count2 = new Map();
  for (const char of word1) {
    count1.set(char, (count1.get(char) || 0) + 1);
  }
  for (const char of word2) {
    count2.set(char, (count2.get(char) || 0) + 1);
  }

  // Compare character sets and counts
  return [...count1.keys()].sort().toString() === [...count2.keys()].sort().toString() &&
         [...count1.values()].sort().toString() === [...count2.values()].sort().toString();
}
